model/roles.go

```diff
@@ -854,3 +854,39 @@ func (r *Role) IsDevRole() bool {
 	}
 	return false
 }
+
+// GetFileSignature returns the SHA1 of a file based on its path and contents
+func GetFileSignature(filePath string) (string, error) {
```

Comment:
If this function is being used, I think we should make it a common utility function and share it between `model/job.go`, `model/package.go`, and this file. Looks like there are similar things being done in all of them.